### DISCUSSION TITLE
什么情况下需要训练文字检测和文本方向分类器模型？

### DISCUSSION THREAD

**Discussion by pcycccccc** [PaddlePaddle/PaddleOCR#13097] [Q&A] [answered]
我针对我的数据集重新训练了两个模型：文字检测模型和文字识别模型
- 测试【文字识别】模型在我crop数据集上的效果，从输出的结果来看是还不错，蛮准确的。
- 测试【文字检测】模型，我不知道是我的训练哪里可能没有搞好，训练出来的模型效果很差，没有检测出目标（这个待排查），但是使用官方提供的推理模型en_PP-OCRv3_det_infer，效果还行，至少比我训练的模型好（但部分目标存在漏检）
我的疑惑：
      从训练后测试的结果来看，我的问题应该是出现在文字检测模型上的训练，（我会再去仔细看一下文字检测模型如何进行微调训练）。但是我不知道我是否该重新训练文本方向分类器模型，因为我训练的图像里，有些文字是有一定的偏转角度的，并非都是绝对的0°和180°。比如下面的图像（这个结果图是由公开的推理模型en_PP-OCRv3_det_infer，推理得到的结果）：
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/fbbf438a-050e-4b28-9a87-a3088a85fea6)
      从结果可以看到，第二行的目标漏检了，但我不确定这个漏检是需要我重新训练文字检测模型，还是重新训练文本方向分类器模型，是否有人能给予建议和看法！感谢！

**Comment by SWHL** [comment, marked as answer]
以下仅是个人经验。
需要训练文字检测？还是文本方向分类？这个需要做一下实验：选取50张这种图像，将文本检测结果、文本方向分类结果都记录下来，看看是哪一步导致没识别出来，从而确定重新训练哪个。

**Reply by pcycccccc** [reply]
get了！！！谢谢大佬
